application-admintools-ui/src/main/resources/AdminTools/Code/AdminToolsJS.xml

```diff
@@ -123,7 +123,89 @@
       <cache>default</cache>
     </property>
     <property>
-      <code>require(['jquery'], function($) {
+      <code>require(['jquery', 'xwiki-meta', 'xwiki-job-runner'], function($, xwikiMeta, JobRunner) {
+  const getResourceURL = function() {
+    let documentReference = XWiki.Model.resolve('AdminTools.Code.HealthCheckResources', XWiki.EntityType.DOCUMENT);
+    return new XWiki.Document(documentReference).getURL('get');
+  };
+
+  const runHealthCheckJob = function() {
+    var data = [];
+    data.push(
+      {name: 'outputSyntax', value: 'plain'},
+      {name: 'sheet', value: 'wiki:AdminTools.Code.HealthCheckResources'},
+      {name: 'action', value: 'healthCheck'},
+      {name: 'form_token', value: xwikiMeta.form_token}
+    );
+    return Promise.resolve(new JobRunner({
+      createStatusRequest: function(jobId) {
+        return {
+          url: getResourceURL(),
+          data: {
+            outputSyntax: 'plain',
+            sheet: 'AdminTools.Code.HealthCheckResources',
+            data: 'jobStatus',
+            jobId: jobId.join('/')
+          }
+        };
+      }
+    }).run(getResourceURL(), data)).catch((reason) => {
+      return Promise.reject(reason);
+    }).finally(() => {
+      $('#admin_tools_job_start').prop("disabled", false);
+    });
+  };
+
+  const reinitialiseProgressBar = function() {
+    $.get(getResourceURL(), function(data) {
+      let progressBar = $(data).find('div.ui-progress');
+      $('#healthCheck .successmessage').replaceWith(progressBar)
+      $('#healthCheck .warningmessage').replaceWith(progressBar)
+      $('#healthCheck .errormessage').replaceWith(progressBar)
```

Comment:
I'm not sure why you replace all of them with the same element, if that's valid you can write it as:

```suggestion
      $('#healthCheck').find('.successmessage, .warningmessage, .errormessage').replaceWith(progressBar);
```

Reply:
I try and replace all of them because I do not know what the message type from the job will be.